cards/product-prominentimage/component.js

```diff
@@ -20,6 +20,10 @@ class product_prominentimageCardComponent extends BaseCard['product-prominentima
     }
     const linkTarget = AnswersExperience.runtimeConfig.get('linkTarget') || '_top';
 
+    //polyfill styling for ie11
+    var someImages = document.querySelectorAll('.HitchhikerProductProminentImage-img');
+    HitchhikerJS.objectFitImages(someImages);
```

Comment:
would it be possible to do the global polyfill for this, and take this out of the specific card components? We probably don't want to run this every render, and this isn't really the responsibility of the card anyway. Maybe we could do it only the legacy bundle

Reply:
I will try, does ie11 only use the legacy bundles?

Reply:
yup!

Reply:
So this can't be in entry-legacy.js because this script is invoked before the html pages are loaded. The polyfill function would requires the src link populated in the img tag then convert to some inline css styling in the tag.

Reply:
Ah ok got it, maybe we can shuffle this off to the base card component to hide some implementation? like maybe we can add a flag to the child card's constructor (or even have a list of cards in the base card component?)
taking it out of the child cards, in whatever way we decide, will also make it easier to tweak forked versions of these cards in the future.

Reply:
Does this polyfill work in a way where it doesn't do anything if you run it on a browser that *does* support object-fit? If so I don't mind running this a few extra times on ie11, in order to improve our non ie-11 cases and keep the cards a little cleaner

Reply:
the polyfill function would check if the element.style contains object-fit. If so, it wouldn't do anything

Reply:
I moved the implementations into the base card component, which check the list of cards that would require this polyfill. Let me know if the approach is good!